From 7cec71f9e9865bd87a2fee363e38b87d69331315 Mon Sep 17 00:00:00 2001
From: Yaoming Qin <b10945@freescale.com>
Date: Mon, 17 Jan 2011 10:54:31 +0800
Subject: [PATCH 112/635] ENGR00137877 [Android_imx51_BBG] Camera: open
 camera/camcorder fai

The root cause is that the system updated, so the sem_init changed and
the api changed.

Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 liboverlay/overlay.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/liboverlay/overlay.cpp b/liboverlay/overlay.cpp
index 444f312..21c28c9 100755
--- a/liboverlay/overlay.cpp
+++ b/liboverlay/overlay.cpp
@@ -316,7 +316,7 @@ static int create_control_shared_data(overlay_control_shared_t **shared)
     p->refCnt = 1;
 
     //Create the sem for control
-    if(sem_init(&p->overlay_sem, 0, 0) != 0){
+    if(sem_init(&p->overlay_sem, 1, 0) != 0){
         OVERLAY_LOG_ERR("Error!init overlay_sem failed");
         munmap(p, size);
         close(fd);
-- 
1.8.0

